src/Proxy/ProxyFactory.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace GraphAware\Neo4j\OGM\Proxy;
+
+use GraphAware\Common\Type\Node;
+use GraphAware\Neo4j\OGM\EntityManager;
+use GraphAware\Neo4j\OGM\Metadata\NodeEntityMetadata;
+use GraphAware\Neo4j\OGM\Metadata\RelationshipMetadata;
+
+class ProxyFactory
+{
+    protected $em;
+
+    protected $classMetadata;
+
+    protected $proxyDir;
+
+    public function __construct(EntityManager $em, NodeEntityMetadata $classMetadata)
+    {
+        $this->em = $em;
+        $this->classMetadata = $classMetadata;
+        $this->proxyDir = $em->getProxyDirectory();
+    }
+
+    public function fromNode(Node $node)
+    {
+        $object = $this->createProxy();
+        $object->__setNode($node);
+        $initializers = [];
+        foreach ($this->classMetadata->getSimpleRelationships() as $relationship) {
+            $initializer = $this->getInitializerFor($relationship);
+            $initializers[$relationship->getPropertyName()] = $initializer;
+        }
+        $object->__setInitializers($initializers);
+
+        return $object;
+    }
+
+    private function getInitializerFor(RelationshipMetadata $relationship) {
+        if (!$relationship->isCollection()) {
+            $initializer = new SingleNodeInitializer($this->em, $relationship, $this->em->getClassMetadata($relationship->getTargetEntity()));
+        }
+
+        return $initializer;
+    }
+
+    protected function createProxy()
+    {
+        $class = $this->classMetadata->getClassName();
+        $proxyClass = $this->getProxyClass();
+        $proxyFile = $this->proxyDir.'/'.$proxyClass.'.php';
+        $methodProxies = $this->getMethodProxies();
+
+        $content = <<<PROXY
+<?php
+
+use GraphAware\\Neo4j\\OGM\\Proxy\\EntityProxy;
+
+class $proxyClass extends $class implements EntityProxy
+{
+    private \$em;
+    private \$initialized = [];
+    private \$intializers = [];
+    private \$node;
+    
+    public function __setNode(\$node)
+    {
+        \$this->node = \$node;
+    }
+    
+    public function __setInitializers(array \$initializers)
+    {
+        \$this->initializers = \$initializers;
+    }
+    
+    public function __initializeProperty(\$propertyName)
+    {
+        \$value = \$this->initializers[\$propertyName]->initialize(\$this->node);
+        \$this->\$propertyName = \$value;
+    }
+    
+    $methodProxies
+}
+
+PROXY;
+
+        $this->checkProxyDirectory();
+        file_put_contents($proxyFile, $content);
+
+        if (!class_exists($proxyClass)) {
+            require $proxyFile;
+        }
+
+        return $this->newProxyInstance($proxyClass);
+
+    }
+
+    protected function getMethodProxies()
+    {
+        $proxies = '';
+        foreach ($this->classMetadata->getRelationships() as $relationship) {
```

Comment:
Im not asking about *when* =) Im asking for a roadmap because I want to know your plans to see if there is anything I could help with. 

I really like this initiative of yours and I would like to see it as **the** neo4j ogm for PHP. That is why I created the Symfony Bundle for you and that is why I want to help out (if I can) to make sure the project gets stable.